supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/JesCacheHitCopyingActor.scala

```diff
@@ -26,46 +31,64 @@ case class JesCacheHitCopyingActor(override val jobDescriptor: BackendJobDescrip
-        jesWorkflowPaths.jesLogFilename
-      )
-      jesFileNames
-    }
-
     val gcsFileSystem = workflowPaths.gcsFileSystemWithUserAuth
+    val destinationCallRootPath: Path = workflowPaths.toJesCallPaths(jobDescriptor.key).callRootPath
+    val sourceCallRootPath: Path = gcsFileSystem.getPath(callOutputFiles.get("callRootPath").get) //Ruchi: the key should aways be present and never be empty, from my understanding
 
-    def copyFilesInGcs() = {
-      val stringFilePaths = cachedJobOutputs map { case (lqn, jobOutput) => lqn -> jobOutput.wdlValue.valueString }
 
-      stringFilePaths foreach {
-        case (lqn, filePath) => if (filePath.isGcsUrl)
+    def copyFiles = {
+      def copyCachedCallOutputFiles(sourceString: String): Unit = {
+        val sourcePath = gcsFileSystem.getPath(sourceString)
+        PathCopier.copy(sourceCallRootPath, sourcePath, destinationCallRootPath)
+      }
+      seqOfSimpletons ++ callOutputFiles foreach {
+        case WdlValueSimpleton(key, wdlFile: WdlFile) =>
+          println(s"the key is: $key and the primitive is: $wdlFile") //Ruchi: For debugging, will remove pre-merge
+          copyCachedCallOutputFiles(wdlFile.value)
+        case(fileName: String, filePath: String) if fileName != "callRootPath" =>
+          println(s"the key is: $fileName and the primitive is: $filePath")
           copyCachedCallOutputFiles(filePath)
+        case debug =>
+          println(s"the other is $debug")
       }
     }
 
-    //nested array output copying? or more complicated file structures
-    def copyCachedCallOutputFiles(sourcePath: String): Unit = {
-      val sourceCallDirectory = gcsFileSystem.getPath(sourcePath)
-      // suspicious that this is unused
-      val buildCallDestinationDirectory = gcsFileSystem.getPathAsDirectory(callRootPath.getParent.toString)
+    def tellMetadata(key: String, value: Any): Unit = {
+      val event = metadataEvent(key, value)
+      val putMetadataAction = PutMetadataAction(event)
+      serviceRegistryActor ! putMetadataAction // and forget
+    }
+
+    def metadataEvent(key: String, value: Any) = {
+      val metadataValue = MetadataValue(value)
+      MetadataEvent(metadataKey(key), metadataValue)
+    }
+
+    def metadataKey(key: String) = MetadataKey(workflowId, Option(metadataJobKey), key)
 
-      PathCopier.copy(callRootPath, sourceCallDirectory.getParent, callRootParent)
+    def updateMetadata = {
+      tellMetadata(CallMetadataKeys.CallRoot, destinationCallRootPath)
+      tellMetadata(JesMetadataKeys.GoogleProject, jesAttributes.project)
+      tellMetadata(JesMetadataKeys.ExecutionBucket, jesAttributes.executionBucket)
+      tellMetadata(JesMetadataKeys.EndpointUrl, jesAttributes.endpointUrl)
     }
 
-    copyFilesInGcs()
-    Future(SucceededResponse(jobDescriptor.key, Option(0), cachedJobOutputs))
+    def createJobOutputs: JobOutputs = WdlValueBuilder.toJobOutputs(jobDescriptor.call.task.outputs, seqOfSimpletons)
+
+   copyFiles
```

Comment:
I've started wrapping obvious Failure modes into the non-retryables, but I would love if you could throw a second glance for any important misses.